chore: first template adjustments

0x-r4bbit opened the pull request:
This is an initial take on adjusting the foundry-template to the Vac's smart contract unit's needs. In a nutshell what this does is:

1. Update README to not refer to upstream repo where not necessary
2. Removes PRBergTest library from dependencies
3. Removes `FUNDING.qml`
4. Adjust CI actions

There are more things to be done and decided on in follow-up commits.

rymnc commented on package.json:
```diff
@@ -1,11 +1,8 @@
-    "name": "Paul Razvan Berg",
-    "url": "https://github.com/PaulRBerg"
-  },
+  "author": {},
```
Maybe add yourself as the author here, and link to https://vac.dev?

0x-r4bbit replied:
Done

rymnc reviewed (approved):
LGTM, minor suggestion

rymnc commented on .github/scripts/rename.sh:
```diff
@@ -31,8 +31,6 @@ sedi () {
-sedi "/gitpod/ s|PaulRBerg/foundry-template|"${GITHUB_REPOSITORY}"|;" "README.md"
-sedi "/gitpod-badge/ s|PaulRBerg/foundry-template|"${GITHUB_REPOSITORY}"|;" "README.md"
+# Rename instances of "vacp2p/foundry-template" to the new repo name in README.md for badges only
 sedi "/gha/ s|PaulRBerg/foundry-template|"${GITHUB_REPOSITORY}"|;" "README.md"
```
```suggestion
sedi "/gha/ s|vacp2p/foundry-template|"${GITHUB_REPOSITORY}"|;" "README.md"
```

rymnc commented on .github/scripts/rename.sh:
```diff
@@ -31,8 +31,6 @@ sedi () {
-sedi "/gitpod-badge/ s|PaulRBerg/foundry-template|"${GITHUB_REPOSITORY}"|;" "README.md"
+# Rename instances of "vacp2p/foundry-template" to the new repo name in README.md for badges only
 sedi "/gha/ s|PaulRBerg/foundry-template|"${GITHUB_REPOSITORY}"|;" "README.md"
 sedi "/gha-badge/ s|PaulRBerg/foundry-template|"${GITHUB_REPOSITORY}"|;" "README.md"
```
```suggestion
sedi "/gha-badge/ s|vacp2p/foundry-template|"${GITHUB_REPOSITORY}"|;" "README.md"
```

0x-r4bbit commented:
@rymnc thank you for your feedback.
I've done the following additional changes:

1. Added your suggested changes in `scripts`
2. Removed some more boilerplate in tests that we don't need for now
3. Added a `PULL_REQUEST_TEMPLATE.md` with an initial (but simple) checklist
4. Generated a first gas snapshot
